src/main/java/org/jabref/gui/openoffice/StyleSelectDialogView.java

```diff
@@ -0,0 +1,138 @@
+        this.dialogService = dialogService;
+        this.loader = loader;
+
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        setResultConverter(button -> {
+            if (button == ButtonType.OK) {
+                viewModel.storePrefs();
+                return tvStyles.getSelectionModel().getSelectedItem().getStyle();
+            }
+            return null;
+
+        });
+        setTitle(Localization.lang("Style selection"));
+    }
+
+    @FXML
+    private void initialize() {
+
+        viewModel = new StyleSelectDialogViewModel(dialogService, loader, preferencesService);
+
+        previewArticle = new PreviewPanel(null, new BibDatabaseContext(), preferencesService.getKeyBindingRepository(), preferencesService.getPreviewPreferences(), dialogService, ExternalFileTypes.getInstance());
+        previewArticle.setEntry(TestEntry.getTestEntry());
+        vbox.getChildren().add(previewArticle);
+
+        previewBook = new PreviewPanel(null, new BibDatabaseContext(), preferencesService.getKeyBindingRepository(), preferencesService.getPreviewPreferences(), dialogService, ExternalFileTypes.getInstance());
+        previewBook.setEntry(TestEntry.getTestEntryBook());
+        vbox.getChildren().add(previewBook);
+
+        colName.setCellValueFactory(cellData -> cellData.getValue().nameProperty());
+        colJournals.setCellValueFactory(cellData -> cellData.getValue().journalsProperty());
+        colFile.setCellValueFactory(cellData -> cellData.getValue().fileProperty());
+        colDeleteIcon.setCellValueFactory(cellData -> cellData.getValue().internalStyleProperty());
+
+        new ValueTableCellFactory<StyleSelectItemViewModel, Boolean>().withGraphic(internalStyle -> {
+            if (!internalStyle) {
+                return IconTheme.JabRefIcons.DELETE_ENTRY.getGraphicNode();
+            }
+            return null;
+
+        }).withOnMouseClickedEvent(item -> {
+            return evt -> viewModel.deleteStyle();
+        }).withTooltip(item -> {
+            return Localization.lang("Remove style");
+        }).install(colDeleteIcon);
+
+        edit.setOnAction(e -> viewModel.editStyle());
+
+        new ViewModelTableRowFactory<StyleSelectItemViewModel>()
+                                                                .withOnMouseClickedEvent((item, event) -> {
+                                                                    if (event.getClickCount() == 2) {
+                                                                        viewModel.viewStyle(item);
+                                                                    }
+                                                                }).withContextMenu(item -> createContextMenu())
+                                                                .install(tvStyles);
+        tvStyles.getSelectionModel().selectedItemProperty().addListener((observable, oldvalue, newvalue) -> {
```

Comment:
why not using a binding?

Reply:
Table has ReadOnlyProperty of selectedItemProperty only